core/templates/pages/oppia-root/routing/app.routing.module.ts

```diff
@@ -79,6 +79,14 @@ const routes: Route[] = [
       .then(m => m.LearnerDashboardPageModule),
     canActivate: [IsLoggedInGuard]
   },
+  {
+    path: AppConstants.
```

Comment:
Should we use `edit-learner-group` over here and in the module use "learnerGroupId:"? Also, this way we won't have to use contextService (which imports lots of other services) in the auth-guard.

Reply:
dropped the use of contextService, Done.

Reply:
Sorry for the missunderstanding but can we do something around:
```
in app.routing do:

{
    path: 'edit-learner-group',
    loadChildren: () => import('....')
    ...
  }
  ```
  
  ```
  In the module config do:
  {
    path: ':learnerGroupId',
    component: ...
  }
  ```
  
  
  Also, we can use `route.paramMap.get('id')` in canActivate to extract id instead of using window. 
  
  WDYT?

Reply:
yes sounds good will try it.

Reply:
Done